hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -1430,4 +1433,26 @@ public void initializeBlockNodeConnections() {
             }
         }
     }
+
+    @Override
+    public Consumer<PlatformEvent> staleEventCallback() {
+        return event -> {
+            final HederaInjectionComponent component = requireNonNull(daggerApp);
+            final Iterator<Transaction> iterator = event.transactionIterator();
+            while (iterator.hasNext()) {
+                final Transaction tx = iterator.next();
+                final Bytes bytes = tx.getApplicationTransaction();
+                try {
+                    final SignedTransaction signedTransaction = SignedTransaction.PROTOBUF.parse(bytes);
+                    final TransactionBody transactionBody =
+                            TransactionBody.PROTOBUF.parse(signedTransaction.bodyBytes());
+                    final TransactionID transactionID = transactionBody.transactionIDOrThrow();
+                    // Mark the TransactionID as stale for resubmission allowance and query reporting.
+                    component.deduplicationCache().markStale(transactionID);
+                } catch (ParseException e) {
+                    // Ignore parsing errors for stale events
```

Comment:
I can't think of how it would happen, but it wouldn't hurt to use `ParseException ignore` here, and another `catch (Exception e)` with a `log.warn()` if something shows up.